code/API_definitions/kyc-tenure.yaml

```
@@ -0,0 +1,486 @@
+
+    # Authorization and authentication
+
+    The "Camara Security and Interoperability Profile" provides details on how a client requests an access token. Please refer to Identify and Consent Management (https://github.com/camaraproject/IdentityAndConsentManagement/) for the released version of the Profile.
+
+    Which specific authorization flows are to be used will be determined during onboarding process, happening between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API, while also being subject to the prevailing legal framework dictated by local legislation.
+
+    It is important to remark that in cases where personal user data is processed by the API, and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of 3-legged access tokens becomes mandatory. This measure ensures that the API remains in strict compliance with user privacy preferences and regulatory obligations, upholding the principles of transparency and user-centric data control.
+
+    # API functionality
+
+    The API defines one service endpoint:
+
+    - `POST /check-tenure`
+
+      Takes the network subscription identifier (e.g. the mobile phone number for a mobile network subscriber) and the specified minimum tenure date to validate for the associated network subscription. This endpoint will respond with a confirmation of whether or not the network subscriber tenure is longer than the specified date timestamp, optionally supplemented with details of the subscription contract type.
+
+    To call this endpoint, the API consumer must first obtain a valid access token with the specified scope from the specified token endpoint, which is then passed to the endpoint via the `Authorization` header. For more details on access token processing, see below.
+
+    ## Inputs
+
+    The endpoint request body is a JSON object with the following parameters:
+    - `tenureDate`: The date from which continuous tenure of the identified network subscriber is required to be confirmed. This field is always required.
+    - `phoneNumber`: The network subscription identifier (i.e. the phone number of the subscriber). This field is only required if no network subscription identifier is associated with the access token.
+
+    ## Outputs
+
+    If successful, a JSON object is returned containing the following data:
+    - `tenureDateCheck`: `true` when the identified network subscription has had valid tenure since `tenureDate`, otherwise `false`
+    - `contractType`: The network subscription account type, if known
+
+    An example of a JSON response object is as follows:
+    ```
+    {
+       "tenureDateCheck": true,
+       "contractType": "PAYM"
+    }
+    ```
+
+    ## Errors
+
+    If the authentication token is not valid, a `401 UNAUTHENTICATED` error is returned
```

I'm not sure about this one, but since both are defined maybe we should mention both

```suggestion
    If the authentication token is not valid or is expired, a `401 UNAUTHENTICATED` or `401 AUTHENTICATION_REQUIRED' error is returned
```

---

I updated the text to include both possible errors. My understanding is that `UNAUTHENTICATED` is when the token is missing or not a valid token (and never was), whereas `AUTHENTICATION_REQUIRED` is used when the token had been valid, but is no longer. I'm not so sure we actually need to distinguish these errors.